pdns/lua-base4.cc

```diff
@@ -17,18 +18,18 @@
 BaseLua4::BaseLua4() {
 }
 
-int BaseLua4::loadFile(const std::string& fname)
+void BaseLua4::loadFile(const std::string& fname)
 {
-  int ret = 0;
   std::ifstream ifs(fname);
   if (!ifs) {
-    ret = errno;
-    SLOG(g_log << Logger::Error << "Unable to read configuration file from '" << fname << "': " << stringerror(ret) << endl,
-         g_slog->withName("lua")->error(Logr::Error, ret, "Unable to read configuration file", "file", Logging::Loggable(fname)));
-    return ret;
+    assert(errno != 0);
```

Comment:
does this assert have a risk of stopping the whole process?

Reply:
or is the point that a failing ifstream that leaves errno zero is a very clear bug, in which case exiting is fine?

Reply:
It's the latter, yes.

Reply:
That's what I thought as well. Though on second thought: while system calls are guaranteed to set `errno` on error, in general lib functions are not. So I prefer the assert to be removed.

Reply:
Alright, done. I fixed-up and force-pushed, I hope that's OK since the change is simple enough.

Reply:
:+1: